tests/unittests/benchmark/task/profet/test_profet_task.py

```diff
@@ -0,0 +1,375 @@
+        if path == REAL_PROFET_DATA_DIR and REAL_PROFET_DATA_DIR.exists():
+            # Return real datasets.
+            logger.info(f"Testing using the real Profet datasets.")
+            return real_load_data(path, benchmark=benchmark)
+        # Generate fake datasets.
+        logger.info(f"Warning: Using random data instead of the actual profet training data.")
+        x_shape, y_shape, c_shape = shapes[benchmark]
+        X = np.random.rand(*x_shape)
+        min_y = y_min[benchmark]
+        max_y = y_max[benchmark]
+        Y = np.random.rand(*y_shape) * (max_y - min_y) + min_y
+        min_c = c_min[benchmark]
+        max_c = c_max[benchmark]
+        C = np.random.rand(*c_shape) * (max_c - min_c) + min_c
+        return X, Y, C
+
+    monkeypatch.setattr(orion.benchmark.task.profet.profet_task, "load_data", _load_data)
+    # NOTE: Need to set the item in the globals because it might already have been imported.
+    monkeypatch.setitem(globals(), "load_data", _load_data)
+
+
+@pytest.mark.timeout(10)
+@requires_profet_data
+@pytest.mark.parametrize("benchmark", ["fcnet", "forrester", "svm", "xgboost"])
+def test_download_fake_datasets(tmp_path_factory, benchmark: str, load_fake_data):
+    # TODO: Downloading the data takes a VERY long time, even though the datasets are relatively
+    # small! Would it be alright to store those datasets somewhere?
+    real_input_dir: Path = REAL_PROFET_DATA_DIR
+    real_x, real_y, real_c = load_data(real_input_dir, benchmark=benchmark)
+
+    fake_input_dir: Path = tmp_path_factory.mktemp("profet_data")
+    fake_x, fake_y, fake_c = load_data(fake_input_dir, benchmark=benchmark)
+    # assert False, (x.dtype, y.dtype, c.dtype)
+    # assert False, (x.dtype, y.dtype, c.dtype)
+    assert real_x.shape == fake_x.shape
+    assert real_y.shape == fake_y.shape
+    assert real_c.shape == fake_c.shape
+
+    assert real_x.dtype == fake_x.dtype
+    assert real_y.dtype == fake_y.dtype
+    assert real_c.dtype == fake_c.dtype
+
+    assert 0 <= real_x.min() and 0 <= fake_x.min()
+    assert real_x.max() <= 1 and fake_x.max() <= 1
+
+    min_y, max_y = y_min[benchmark], y_max[benchmark]
+    assert min_y <= real_y.min() and min_y <= fake_y.min()
+    assert real_y.max() <= max_y and fake_y.max() <= max_y
+
+    min_c, max_c = c_min[benchmark], c_max[benchmark]
+    assert min_c <= real_c.min() and min_c <= fake_c.min()
+    assert real_c.max() <= max_c and fake_c.max() <= max_c
+
+
+class ProfetTaskTests:
+    """ Base class for testing Profet tasks. """
+
+    # TODO: What would be a good set of tests for the profet tasks?
+
+    Task: ClassVar[Type[ProfetTask]]
+
+    @pytest.mark.timeout(30)
+    def test_instantiating_task(
+        self,
+        profet_train_config: MetaModelTrainingConfig,
+        profet_input_dir: Path,
+        tmp_path_factory,
+    ):
+        """ TODO: Test that when instantiating multiple tasks with the same arguments, the
+        meta-model is trained only once.
```

Comment:
I would also mock the top function used to train the meta-model to test that is was indeed called the first time but not the second time. Also, the tests for the arguments should be moved into a separate unit-test.

Reply:
Fixed.

Reply:
Just to make sure I am not confused here, get_task_network is called only if we have no checkpoint, so it is trained, and load_task_network is called when we have a checkpoint. Am I understanding correctly?

Reply:
Yes, that's it.

Reply:
ok, thanks!